prepline_general/api/vantiq/chunking.py

```diff
@@ -0,0 +1,46 @@
+from typing import Iterable, Iterator, Optional
+
+from unstructured.chunking.base import BoundaryPredicate, is_on_next_page, PreChunker
+# noinspection PyProtectedMember
+from unstructured.chunking.title import _ByTitleChunkingOptions
+from unstructured.documents.elements import Element
+from unstructured.utils import lazyproperty
+
+
+class _ByPageChunkingOptions(_ByTitleChunkingOptions):
+    @lazyproperty
+    def boundary_predicates(self) -> tuple[BoundaryPredicate, ...]:
+        """The semantic-boundary detectors to be applied to break pre-chunks.
+
+        For the `by_title` strategy these are sections indicated by a title (section-heading), an
+        explicit section metadata item (only present for certain document types), and optionally
+        page boundaries.
+        """
+
+        def iter_boundary_predicates() -> Iterator[BoundaryPredicate]:
+            yield is_on_next_page()
+
+        return tuple(iter_boundary_predicates())
+
+
+def chunk_by_page(
+    elements: Iterable[Element],
+    *,
+    combine_text_under_n_chars: Optional[int] = None,
+    include_orig_elements: Optional[bool] = None,
+    max_characters: Optional[int] = None,
+    new_after_n_chars: Optional[int] = None,
+    overlap: Optional[int] = None,
+    overlap_all: Optional[bool] = None,
+) -> list[Element]:
+    opts = _ByPageChunkingOptions.new(
+        combine_text_under_n_chars=combine_text_under_n_chars,
+        include_orig_elements=include_orig_elements,
+        max_characters=max_characters,
+        new_after_n_chars=new_after_n_chars,
+        overlap=overlap,
+        overlap_all=overlap_all,
+    )
+    pre_chunks = PreChunker.iter_pre_chunks(elements, opts)
```

Comment:
Tried using the same chunking at `by_title` (with the tweaked options), but it ends up merging elements across the page boundaries (which we don't want).